kwc-challenge-set.html

```diff
@@ -8,27 +12,469 @@
+
+            <div class="circle-box">
+                <div class="circle-content">
+                    <div id="canvas"></div>
+                    <div id="progress"></div>
+                    <iron-image src="[[_computeCompleteImage(_completed)]]"></iron-image>
+                </div>
+            </div>
+
+            <div class="control-box">
+                <p class="control-title">
+                    [[challengeSet.title]]
+                </p>
+                <p class$="[[_computeCompleteClass(_completed)]]">
+                    [[_computeProgressText(progress, _completed)]]
+                </p>
+                <p class="control-content">
+                    [[challengeSet.description]]
+                </p>
+                <kwc-button class="control-button"
+                    hidden$="[[_completed]]"
+                    on-tap="_openNextChallenge"
+                >
+                    continue
+                </kwc-button>
+                <kwc-button class="control-button"
+                    hidden$="[[!_completed]]"
+                    on-tap="_openFirstChallenge"
+                >
+                    redo
+                </kwc-button>
+            </div>
+        </div>
+    </template>
+    <script>
+        Polymer({
+            is: 'kwc-challenge-set',
+            properties: {
+                /*
+                 * The data for the challenge set to build a progress bar for.
+                 * @type {Object}
+                 */
+                challengeSet: {
+                    type: Object,
+                    value: () => {
+                        return {};
+                    }
+                },
+                /*
+                 * A number representing the number of challenges in this set that are completed.
+                 * @type {Number}
+                 */
+                progress: {
+                    type: Number,
+                    value: 0
+                },
+                _nextChallenge: {
+                    type: Object,
+                    value: () => {
+                        return {};
+                    }
+                },
+                _numNodes: {
+                    type: Number,
+                    computed: '_computeNumNodes(challengeSet)'
+                },
+                /*
+                 * The radius of each challenge progress circle in the progress bar.
+                 */
+                radius: {
+                    type: Number,
+                    value: 125
+                },
+                /*
+                 * The url of the current user's avatar. If none given a default icon is used.
+                 */
+                avatarUrl: {
+                    type: String,
+                    value: null
+                },
+                _avatarUrl: {
+                    type: String,
+                    computed: '_computeIconUrl("avatar", avatarUrl)'
```

Comment:
Seeing how you've done it, I wonder if you could store those data on a js variable on a separate file and just use it as a static config, let's say:

```
let assets = {
  "avatar": "data:image whatever",
  "complete": "other stuff"
}
```

and use

```
_avatarUrl: {
  type: String,
  value: () => assets.avatar
}
```

But that is just sugar ;P